plugins/org.komodo.shell/src/org/komodo/shell/commands/core/ServerCommand.java

```diff
@@ -0,0 +1,687 @@
+                TeiidInstance teiidInstance = teiid.getTeiidInstance(wsStatus.getTransaction());
+                try {
+                    teiidInstance.connect();
+                } catch (Exception ex) {
+                    print(CompletionConstants.MESSAGE_INDENT, Messages.getString(Messages.ServerCommand.ConnectionError,ex.getLocalizedMessage()));
+                }
+                
+                boolean connected = teiidInstance.isConnected();
+                String connectStatus = connected ? Messages.getString(Messages.ServerCommand.Connected) : Messages.getString(Messages.ServerCommand.NotConnected); 
+                print(CompletionConstants.MESSAGE_INDENT, Messages.getString(Messages.ServerCommand.TeiidStatus, teiid.getName(wsStatus.getTransaction()), connectStatus)); 
+                return connected;
+            // Sub-command which disconnects from the default server
+        	} else if (SUBCMD_DISCONNECT.equalsIgnoreCase(subcmdArg)) {
+                
+                print(CompletionConstants.MESSAGE_INDENT, Messages.getString(Messages.ServerCommand.AttemptingToDisconnect,teiid.getName(wsStatus.getTransaction())));
+                TeiidInstance teiidInstance = teiid.getTeiidInstance(wsStatus.getTransaction());
+                if(!teiidInstance.isConnected()) {
+                    print(CompletionConstants.MESSAGE_INDENT, Messages.getString(Messages.ServerCommand.NoServerToDisconnectMsg));
+                }
+                
+                teiidInstance.disconnect();
+                print(CompletionConstants.MESSAGE_INDENT, Messages.getString(Messages.ServerCommand.DisconnectSuccessMsg, teiid.getName(wsStatus.getTransaction())));
+                return true;
+            // Sub-command for showing objects and object details for the connected server
+            } else if (SUBCMD_SHOW.equalsIgnoreCase(subcmdArg)) {
+                String serverObjType = requiredArgument(1, Messages.getString(Messages.ServerCommand.InvalidArgMsg_ServerObjType));
+
+                if(!isTeiidConnected(wsStatus)) {
+                    print( MESSAGE_INDENT, Messages.getString( Messages.ServerCommand.ServerNotConnected ) ); 
+                    return false;
+                }
+                
+                if(!SERVER_OBJ_TYPES_PLURAL.contains(serverObjType.toLowerCase()) && !SERVER_OBJ_TYPES_SINGULAR.contains(serverObjType.toLowerCase())) {
+                    print( MESSAGE_INDENT, Messages.getString( Messages.ServerCommand.InvalidServerObjectType, serverObjType ) ); 
+                    return false;
+                }
+                
+                // Plural type - shows names of all
+                if(SERVER_OBJ_TYPES_PLURAL.contains(serverObjType.toLowerCase())) {
+                    printServerObjectNames(wsStatus,serverObjType);
+                }
+                
+                // Singular type - requires name of the object, then shows object properties
```

Comment:
201 - 204 can be changed to: `} else {` as I think you've already checked to make sure `serverObjType` is contained in `SERVER_OBJ_TYPES_SINGULAR`.